src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java

```diff
@@ -4754,11 +4754,11 @@ private void addJavadocOptions( File javadocOutputDirectory,
                             addArgIfNotEmpty( arguments, "--patch-module", moduleName + '='
                                 + JavadocUtil.quotedPathArgument( getSourcePath( projectSourcepaths.getValue() ) ) );
                             
-                            Files.createDirectory( moduleSourceDir.resolve( moduleName ) );
+                            Files.createDirectories( moduleSourceDir.resolve( moduleName ) );

```

Comment:
The semantics of this method are different from the ones of createDirectory.
If you only want to create the last path you should use
if !Files.isDirectory...createDirectory

Reply:
That's probably better: `moduleSourceDir` should already exist and `moduleName` should reflect 1 directory.

Reply:
Done. I've reworked the fix in terms of `if !Files.isDirectory...createDirectory`.